legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSQLQueryAction.java

```diff
@@ -109,6 +137,12 @@ public RestChannelConsumer prepareRequest(SQLQueryRequest request, NodeClient no
       if (request.isExplainRequest()) {
         LOG.info("Request is falling back to old SQL engine due to: " + e.getMessage());
       }
+
+      /**
+       * Setting errorStr member variable is used to aggregate error messages when both legacy and new SQL engines fail.
+       * This implementation can be removed when the legacy SQL engine is deprecated.
+       */
+      setErrorStr(ErrorMessageFactory.createErrorMessage(e, isClientError(e) ? BAD_REQUEST.getStatus() : SERVICE_UNAVAILABLE.getStatus()).toString());
```

Comment:
what if request.isSupported==false? Should we capture it also?

Reply:
That's a fair point, I can revise to set the error string when request is not supported.